scripts/quests/bastok/Escort_For_Hire_Bastok.lua

```diff
@@ -0,0 +1,257 @@
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            player:getQuestStatus(xi.questLog.WINDURST, xi.quest.id.windurst.ESCORT_FOR_HIRE) ~= xi.questStatus.QUEST_ACCEPTED and
+            player:getQuestStatus(xi.questLog.SANDORIA, xi.quest.id.sandoria.ESCORT_FOR_HIRE) ~= xi.questStatus.QUEST_ACCEPTED and
+            player:getFameLevel(xi.quest.fame_area.BASTOK) >= 6 and
+            player:getCharVar('ESCORT_CONQUEST') < NextConquestTally()
+        end,
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Trilok'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasCompletedQuest(xi.questLog.BASTOK, xi.quest.id.bastok.ESCORT_FOR_HIRE) then
+                        quest.firstTimeCompletion = false
+                    end
+
+                    return quest:progressEvent(291) -- start
+                end,
+            },
+
+            onEventFinish =
+            {
+                [291] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.ROLANBERRY_FIELDS] =
+        {
+            onTriggerAreaEnter =
+            {
+                [1] = function(player, triggerArea)
+                    if quest:getVar(player, 'Prog') >= 0 and quest:getVar(player, 'Prog') <= 2 then
+                        quest:setVar(player, 'Front', 1)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.CRAWLERS_NEST] =
+        {
+            onTriggerAreaLeave =
+            {
+                [1] = function(player, triggerArea)
+                    quest:setVar(player, 'Front', 0)
+                end
+            },
+
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    if
+                        quest:getVar(player, 'Front') == 1 and
+                        GetServerVariable('[Escort]Olavia') == 0 and
+                        not player:hasKeyItem(xi.ki.COMPLETION_CERTIFICATE)
+                    then
+                        if quest:getVar(player, 'Prog') == 0 then
+                            return 6
+                        elseif quest:getVar(player, 'Prog') == 2 then
+                            return 6
+                        end
+                    end
+                end,
+            },
+
+            onEventUpdate =
+            {
+                [6] = function(player, csid, option, npc)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [6] = function(player, csid, option, npc)
+                    local function spawnEscort(escort, checkEscort, zone, playerPos)
+                        local olavia = zone:insertDynamicEntity({
```

Comment:
is Olavia etc actually a DE on retail? curious

Reply:
Wtb capture of this to verify

Reply:
![image](https://github.com/LandSandBoat/server/assets/60417494/aa683019-8e4c-4272-a0ca-58f1b3161a60)
huh. no shit. they actually used a DE on retail.

Reply:
![image](https://github.com/LandSandBoat/server/assets/60417494/167deca5-a08a-4f9a-8e95-866fca5bf4cf)

Reply:
Speed 60 despite her not running fast. 60 is more than my own player..